### ISSUE TITLE
Autoscaling

### ISSUE THREAD

**Issue by pritchardjonathan** [peervpn/peervpn#10]
How would you tackle an autoscaling scenario where you want to be able to dynamically add and remove nodes to and from a L3 mesh. As far as I can see config requires a static IP to be defined per node.

**Comment by miolini** [comment]
You could switch to https://github.com/gophergala2016/meshbird

**Comment by mstenz** [comment]
only one node need a static ip, and as long as the network exists by minimum 2 nodes this node can also disappear because all the peers are exchanged within the protocol. If you really want to have a dynamically peer network best is setup one static server that is always running with fixed ip (can be ordered from 3$ per month) and all others can be dynamic (for example behind ADSL lines at home).

**Comment by pritchardjonathan** [comment]
So would I just leave out the ifconfig4 line for the non-static nodes?

**Comment by mstenz** [comment]
No, as ifconfig4 line specify the VPN ipaddress for this node to use within the tunnel, without an ipaddress the different nodes cannot communicate.

**Comment by wankdanker** [comment]
Would it be possible to use dhclient on the VPN interface if there was a DHCP server somewhere on the VPN? That would be interesting.

**Comment by mstenz** [comment]
I have not tried it, but I don't think this is possible.

**Comment by wankdanker** [comment]
peervpn-dnsmasq.conf
```
interface dnsmasq0
ifconfig4 10.10.10.1/24
upcmd dnsmasq -i dnsmasq0 --dhcp-range=10.10.10.10,10.10.10.20,255.255.255.0,12h
port 5678
```

peervpn-dhclient.conf
```
initpeers 127.0.0.1 5678
interface dhclient0
upcmd dhclient -nw dhclient0
```

Apparently, DHCP will work over the VPN. So, you could have a static head node (in this example defined in peervpn-dhclient.conf with initpeers 127.0.0.1 5678) that serves up DHCP addresses.

So, you would only need to specify a static address for one node (so dnsmasq can bind to an address) and each other node would need to know at least the head node's (or another node's) address.

This was tested on a local machine but, I could see the DHCP working with tcpdump on the vpn interfaces.

**Comment by McL0v1n** [comment]
Thank you for this, I will test out the dhclient on my setup. If this is confirmed for me, it should be added into the tutorial as an option.

**Comment by maurociancio** [comment]
This worked excellent!
